utility/fetchClientId.js

```diff
@@ -0,0 +1,103 @@
+/**	Utility function that fetches client IDs (GA3, GA4, Segment, Facebook) from the browser
+ * and sends them asyncrhonously to the Littledata server.
+ * You should run this function when all the tracking libraries have been loaded and when you have
+ * the Shopfy cartToken or checkoutToken created by the Shopify API.
+ *
+ * @param {*} shopifyToken Shopify Cart token or Shopify Checkout token
+ * @param {*} ga3PropertyId GA3 Property ID (ex. UA-XXXX-X)
+ * @param {*} ga4MeasurementId GA4 Measurement ID (ex. G-XXXX)
+ */
+
+const fetchClientId = async ({
+	shopifyToken,
+	ga3PropertyId,
+	ga4MeasurementId,
+}) => {
+	if (!shopifyToken) {
+		return
+	}
+	const ldPayload = {}
+
+	const getGaIds = async (id, type) => {
+		const getCliendId = new Promise(resolve => {
+			gtag('get', id, type, resolve)
```

Comment:
If GTag is being adblocked, this should fail at interpret time, since gtag is not defined.

Reply:
this is not the latest code